python/ql/src/semmle/python/frameworks/Django.qll

```diff
@@ -1987,7 +1995,18 @@ private module Django {
    */
   private DataFlow::Node djangoRouteHandlerFunctionTracker(DataFlow::TypeTracker t, Function func) {
     t.start() and
-    result = DataFlow::exprNode(func.getDefinition())
+    (
+      not exists(func.getADecorator()) and
+      result.asExpr() = func.getDefinition()
+      or
+      // If the function has decorators, we still want to model the function as being
+      // the request handler for a route setup. In such situations, we must track the
+      // last decorator call instead of the function itself.
+      //
+      // Note that this means that we blindly ignore what the decorator actually does to
+      // the function, which seems like an OK tradeoff.
```

Comment:
Should we at least check that the decorator has anything to do with request handling (and possibly Django)?
We could for instance have a list of function names to compare with.

Reply:
I would rather not. In the real-code base that motivated this change, it was a decorator from a very small third-party library that we would probably never invest time in manually modeling.

I'm optimistic about this change, and think it'll just work out. If that turns out to be wrong, I'm very happy to change my mind. 

However, I think allowing any decorator align with our current goals of being able to find more results, and accepting that there might be extra FPs,

Reply:
Allright, ship to learn 🚀